sky/clouds/gcp.py

```diff
@@ -402,6 +408,20 @@ def get_feasible_launchable_resources(self, resources):
                     cpus = float(resources.cpus)
                     if cpus != num_cpus_in_tpu_vm:
                         return ([], fuzzy_candidate_list)
+            memory_in_tpu_vm = 624
+            if resources.memory is not None:
+                if resources.memory.endswith('+'):
+                    memory = float(resources.memory[:-1])
+                    if memory > memory_in_tpu_vm:
+                        return ([], fuzzy_candidate_list)
+                elif resources.memory.endswith(('x', 'X')):
```

Comment:
As it is only used internally, I think we can only allow 'x' and disallow 'X'. WDYT?